ui/packages/lib/src/providers/project/context.js

```diff
@@ -0,0 +1,62 @@
+import React, { useContext, useMemo } from "react";
+import { ApplicationsContext } from "../application";
+import { useMlpApi } from "../../hooks/useMlpApi";
+import { useMatch } from "react-router-dom";
+import urlJoin from "proper-url-join";
+
+const projectIdKey = "lastSelectedProjectId";
+const getSelectedProjectId = (projectId, projects) => {
+  if (!!projectId) {
+    localStorage.setItem(projectIdKey, projectId);
```

Comment:
General question - we save the last accessed project info only to the client side `localStorage` as opposed to server side - is that correct? Are we considering changing this to server-side some day?

Reply:
This isn't a new piece, the code was just moved from the `currentProjectContext.js` file. But answering the question if this will ever be migrated to server side – maybe one day, when we have a critical mass of the other user preferences that makes sense to store between the different browser sessions. But for now it's hard to justify spending time on this.